# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.cpp
# Commit: 5a94b945a3ad
# Full Hash: 5a94b945a3ad6bf603d47c07788b7a70ba651320
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 4: Implement ActorDestroy r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227689
# ==============================================================================

diff -r 984f6e687ff5 -r 5a94b945a3ad dom/notification/NotificationParent.cpp
--- a/dom/notification/NotificationParent.cpp	Thu Nov 07 17:32:04 2024 +0000
+++ b/dom/notification/NotificationParent.cpp	Thu Nov 07 17:32:05 2024 +0000
@@ -6,6 +6,7 @@
 
 #include "NotificationParent.h"
 
+#include "NotificationUtils.h"
 #include "mozilla/ipc/Endpoint.h"
 
 namespace mozilla::dom::notification {
@@ -40,4 +41,18 @@
   return NS_OK;
 }
 
+void NotificationParent::ActorDestroy(ActorDestroyReason aWhy) {
+  nsAutoString alertName;
+  GetAlertName(alertName);
+  UnregisterNotification(mPrincipal, mId, alertName, CloseMode::InactiveGlobal);
+}
+
+void NotificationParent::MaybeInitAlertName() {
+  if (!mAlertName.IsEmpty()) {
+    return;
+  }
+
+  ComputeAlertName(mPrincipal, mOptions.tag(), mId, mAlertName);
+}
+
 }  // namespace mozilla::dom::notification